mteb/abstasks/AbsTask.py

```diff
@@ -12,6 +13,7 @@
 
 class AbsTask(ABC):
     metadata: TaskMetadata
+    max_n_samples = 2048
```

Comment:
@isaac-chung I believe this line is problematic as it can be understood as the max number of samples for the dataset (which is not actually enforced). I would instead move it to `stratified_subsampling` as an argument with a default.